types/bluebird/index.d.ts

```diff
@@ -566,7 +566,7 @@ declare class Bluebird<R> implements PromiseLike<R>, Bluebird.Inspection<R> {
   /**
    * Same as calling `Promise.all(thisPromise)`. With the exception that if this promise is bound to a value, the returned promise is bound to that value too.
    */
-  all(this: Bluebird<Iterable<{}>>): Bluebird<R>;
+  all<Q>(this: Bluebird<R & Iterable<Q>>): Bluebird<R>;
```

Comment:
I think having no generic is probably a little better since the generic is only used in one place in the signature. However, as you mentioned in your description, this is consistent with the other functions in this class.

Reply:
If you have some other way to make the new tests pass without needing generics that would be fine by me, but I don't know of one off the top of my head.